test/integration/logbucket-project/logbucket_project_test.go

```diff
@@ -0,0 +1,96 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logbucket_project
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	defaultRetentionDays int64 = 30
+)
+
+func TestLogBucketProjectModule(t *testing.T) {
+
+	bpt := tft.NewTFBlueprintTest(t,
+		tft.WithTFDir("../../../examples/logbucket/project"),
+	)
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		bpt.DefaultVerify(assert)
+
+		for _, tc := range []struct {
+			projId             string
+			bktName            string
+			sinkDest           string
+			sinkProjId         string
+			sinkName           string
+			sinkWriterIdentity string
+		}{
+			{
+				projId:             bpt.GetStringOutput("log_bucket_project"),
+				bktName:            bpt.GetStringOutput("log_bucket_name"),
+				sinkDest:           bpt.GetStringOutput("log_sink_destination_uri"),
+				sinkProjId:         bpt.GetStringOutput("log_sink_project_id"),
+				sinkName:           bpt.GetStringOutput("log_sink_resource_name"),
+				sinkWriterIdentity: bpt.GetStringOutput("log_sink_writer_identity"),
+			},
+			{
+				projId:             bpt.GetStringOutput("log_bkt_same_proj"),
+				bktName:            bpt.GetStringOutput("log_bkt_name_same_proj"),
+				sinkDest:           bpt.GetStringOutput("log_sink_dest_uri_same_proj"),
+				sinkProjId:         bpt.GetStringOutput("log_sink_id_same_proj"),
+				sinkName:           bpt.GetStringOutput("log_sink_resource_name_same_proj"),
+				sinkWriterIdentity: "",
+			},
+		} {
+			//************************
+			// Assert bucket details *
+			//************************
+			bktFullName := fmt.Sprintf("projects/%s/locations/%s/buckets/%s", tc.projId, "global", tc.bktName)
+			logBucketDetails := gcloud.Runf(t, fmt.Sprintf("logging buckets describe %s --location=%s --project=%s", tc.bktName, "global", tc.projId))
+
+			// assert log bucket name, retention days & location
+			assert.Equal(bktFullName, logBucketDetails.Get("name").String(), "log bucket name should match")
+			assert.Equal(defaultRetentionDays, logBucketDetails.Get("retentionDays").Int(), "retention days should match")
+
+			logSinkDetails := gcloud.Runf(t, fmt.Sprintf("logging sinks describe %s --project=%s", tc.sinkName, tc.sinkProjId))
+
+			// assert log sink name, destination & filter
+			assert.Equal(tc.sinkDest, logSinkDetails.Get("destination").String(), "log sink destination should match")
+			assert.Equal("resource.type = gce_instance", logSinkDetails.Get("filter").String(), "log sink filter should match")
+			assert.Equal(tc.sinkWriterIdentity, logSinkDetails.Get("writerIdentity").String(), "log sink writerIdentity should not be empty")
```

Comment:
"log sink writerIdentity should not be empty" ??
it is empty in the second case.
maybe "should have excepted  log sink writerIdentity" ?

Reply:
I keep "log sink writerIdentity should match" and wrote a comment on the empty writerIdentity example:
```
sinkName:   bpt.GetStringOutput("log_sink_resource_name_same_proj"),
// writerIdentity: As sink and bucket are in same project no service account is needed and writerIdentity is empty
```